File: frontend/src/services/enrollmentService.js
import api from './api';

const enrollmentService = {
    getAll: () => api.get('/learning/enrollments'),
    getById: (id) => api.get(`/learning/enrollments/${id}`),
    getByEmployeeId: (employeeId) => api.get(`/learning/enrollments/employee/${employeeId}`),
    create: (data) => api.post('/learning/enrollments', data),
    update: (id, data) => api.put(`/learning/enrollments/${id}`, data),
    delete: (id) => api.delete(`/learning/enrollments/${id}`),
};

export default enrollmentService;

File: skillsphere-learning-service/src/main/java/com/infosis/nexus/learning/service/EnrollmentService.java
package com.infosis.nexus.learning.service;

import com.infosis.nexus.learning.dto.EnrollmentRequest;
import com.infosis.nexus.learning.dto.EnrollmentResponse;

import java.util.List;

public interface EnrollmentService {
    EnrollmentResponse create(EnrollmentRequest request);
    EnrollmentResponse getById(Long id);
    List<EnrollmentResponse> getAll();
    List<EnrollmentResponse> getByEmployeeId(Long employeeId);
    EnrollmentResponse update(Long id, EnrollmentRequest request);
    void delete(Long id);
}